pkgs/dartpad_ui/lib/model/model.dart

```diff
@@ -270,7 +287,20 @@ class AppServices {
   }
 
   Future<VersionResponse> populateVersions() async {
-    final version = await services.version();
+    VersionResponse version;
+
+    if (useWebsockets) {
+      // Since using websockets is a compile-time config option, and it's only
+      // used for the version call, we create it here on demand.
+      webSocketServices ??= await WebsocketServicesClient.connect(
```

Comment:
I would probably still prefer to create this object in the AppServices constructor (or not, depending on the useWebsockets flag) so that this code is easier to migrate in the future

Reply:
OK, sounds good - creating a websocket connection eagerly, but only if we have the compile time `useWebsockets` const set.